packages/web-frontend/src/historyNavigation/historyMiddleware.js

```diff
@@ -62,14 +63,17 @@ const reactToLocationChange = (store, location, previousLocation) => {
 
   const previousComponents = decodeLocation(previousLocation);
 
-  dispatch(setOverlayComponent(null));
-
   const setComponentIfUpdated = (componentKey, setComponent) => {
     const component = otherComponents[componentKey];
     if (component && component !== previousComponents[componentKey])
       dispatch(setComponent(component));
   };
 
+  if (PROJECTS_WITH_LANDING_PAGES[otherComponents[URL_COMPONENTS.PROJECT]]) {
+    dispatch(setOverlayComponent(PROJECT_LANDING));
+  } else {
+    dispatch(setOverlayComponent(null));
+  }
```

Comment:
A detail, but I think this makes easier to see what is the common logic:

```suggestion
  const isLandingPageProject = PROJECTS_WITH_LANDING_PAGES[otherComponents[URL_COMPONENTS.PROJECT]];
  const overlayComponent = isLandingPageProject ? PROJECT_LANDING : null;
  dispatch(setOverlayComponent(overlayComponent));
```